src/esrs/aggregate.rs

```diff
@@ -32,43 +32,29 @@ pub trait Eraser<
     async fn delete(&self, aggregate_id: Uuid) -> Result<(), Error>;
 }
 
-/// The AggregateManager is responsible for loading an aggregate from the store, mapping commands to events, and
-/// persisting those events in the store. Be careful when implementing this trait, as you will be responsible for
-/// threading AggregateState/Commands/Events correctly.  For example, a bad implementation could result in an AggregateState
-/// that could not be replicated on load.
+/// The AggregateManager is responsible for coupling the Aggregate with a Store, so that the events
+/// can be persisted when handled, and the state can be reconstructed by loading and apply events sequentially.
 ///
-/// Unless you need to perform side effects as part of your command handling/verification you should implement the
-/// safer `Aggregate` trait instead.
+/// It comes batteries-included, as you only need to implement the `event_store` getter. The basic API is:
+/// 1. execute_command
+/// 2. load
+/// The other methods are used internally, but can be overridden if needed.
 #[async_trait]
-pub trait AggregateManager: Identifier {
-    type State: Default + Clone + Debug + Send + Sync;
-    type Command: Send + Sync;
-    type Event: Serialize + DeserializeOwned + Send + Sync;
-    type Error: Send + Sync;
-
+pub trait AggregateManager: Identifier + Aggregate {
     /// Returns the event store, configured for the aggregate
     fn event_store(&self) -> &(dyn EventStore<Self::Event, Self::Error> + Send + Sync);
 
-    /// This function applies the event onto the aggregate and returns a new one, updated with the event data
-    fn apply_event(id: &Uuid, state: Self::State, event: &StoreEvent<Self::Event>) -> Self::State;
-
-    /// Validation should reject any command is inconsistent with the current aggregate state, or would result
-    /// in one or more events that could not be applied onto the aggregate state.
-    fn validate_command(aggregate_state: &AggregateState<Self::State>, cmd: &Self::Command) -> Result<(), Self::Error>;
-
-    async fn do_handle_command(
-        &self,
-        aggregate_state: AggregateState<Self::State>,
-        cmd: Self::Command,
-    ) -> Result<AggregateState<Self::State>, Self::Error>;
-
-    async fn handle_command(
+    /// Validates and handles the command onto the given state, and then persists the events in the store.
+    async fn execute_command(
```

Comment:
Totally dislike the `execute_`. Why is not handle_command like before? Moreover you wrote "Validates and `handles` the command" :joy: